gateway-spi/src/main/java/org/apache/knox/gateway/services/security/token/TokenStateService.java

```diff
@@ -27,22 +27,41 @@
 
   String CONFIG_SERVER_MANAGED = "knox.token.exp.server-managed";
 
+  /**
+   * @return The default duration (in milliseconds) for which a token's life will be extended when it is renewed.
+   */
+  long getDefaultRenewInterval();
+
+  /**
+   * @return The default maximum lifetime duration (in milliseconds) of a token.
+   */
+  long getDefaultMaxLifetimeDuration();
+
   /**
    * Add state for the specified token.
    *
    * @param token     The token.
    * @param issueTime The time the token was issued.
    */
-  void addToken(JWTToken token, long issueTime);
-
+  void addToken(JWTToken token, Long issueTime);
```

Comment:
Is there a reason that all the `long` primitives were changed to `Long` objects? I didn't seem an obvious reason why this was necessary. The `long` primitive should be fine?

Reply:
I see this happened basically everywhere but only commenting once here.

Reply:
I made that change before subsequently exposing the methods for getting the default renewal interval and maximum lifetime values. The Long vs long was so that a null could be passed in, indicating that the default should be applied. Now, clients have to pass in a value, which they can get from the TokenStateService itself, so the params could be reverted to the long type again.

Reply:
The only concern I had was performance based since `Long` is heavier than `long`. I don't know how often these apis are called, but if possible primitive seemed to make more sense when reading through.

Reply:
I agree, and as I've mentioned, it was an iterative change which is no longer needed. I'll revert that, and submit an update to this PR.